public_root_of_trust.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	sigtuf "github.com/sigstore/sigstore/pkg/tuf"
+	tufclient "github.com/theupdateframework/go-tuf/client"
+)
+
+var globalBuffer bytes.Buffer
+
+type inMemoryDest struct{}
+
+func (d inMemoryDest) Write(p []byte) (n int, err error) {
+	return globalBuffer.Write(p)
+}
+
+func (d inMemoryDest) Delete() error {
+	panic("inMemoryDest delete function should not run")
+}
+
+func GetTargets(usage sigtuf.UsageKind, proxy Proxy) ([]sigtuf.TargetFile, error) {
+	// client initialization
+	var httpClient *http.Client
+	if proxy.URL != "" {
+		transport := proxy.HttpTransport()
+		httpClient = &http.Client{Transport: &transport}
```

Comment:
There is no timeout value specified in http.Client object.
Please use the timeout value scanner uses.